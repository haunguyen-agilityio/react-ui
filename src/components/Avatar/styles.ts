// lib MUI
import { Avatar } from '@mui/material'
import { styled } from '@mui/system'

// theme
import theme from '../../themes'

export const StyledAvatar = styled(Avatar)({
  backgroundColor: '#21d083',
  fontWeight: 600,
  color: theme.palette.common.white,
})

export const SmallAvatar = styled(Avatar)({
  backgroundColor: '#aaaaaa',
})
