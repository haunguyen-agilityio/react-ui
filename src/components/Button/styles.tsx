// libraries
import { styled } from '@mui/system'

// components
import { Button } from '@mui/material'

// theme
import typography from '../../themes/typography'

const StyledButton = styled(Button)(({ theme }) => ({
  margin: theme.spacing(),
  padding: `${theme.spacing(1.7)} ${theme.spacing(3)}`,
  height: '38px',
  borderRadius: 0,
  boxShadow: 'none',
  textTransform: 'capitalize',
  fontSize: typography.h4.fontSize,
  color: theme.palette.common.white,
  '&:hover': {
    boxShadow: 'none',
  },
  '&.Mui-disabled': {
    color: theme.palette.common.black,
    backgroundColor: theme.palette.primary.main,
    opacity: 0.5,
  },
  '&.MuiButton-containedSecondary': {
    color: theme.palette.common.black,
    backgroundColor: '#eaeaea',
    border: '1px solid #cbcbcb',
    '&:hover': {
      backgroundColor: '#d0d0d0',
    },
    '&.Mui-disabled': {
      backgroundColor: '#eaeaea',
    },
  },
  '.MuiSvgIcon-root': {
    width: '12px',
    height: '12px',
  },
}))

export default StyledButton
