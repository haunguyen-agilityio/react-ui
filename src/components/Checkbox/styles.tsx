// libraries
import { Checkbox, CheckboxProps, FormControlLabel } from '@mui/material'
import { styled } from '@mui/system'

export const StyledFormControlLabel = styled(FormControlLabel)(({ theme }) => ({
  '.MuiFormControlLabel-label.Mui-disabled': {
    color: theme.palette.common.black,
  },
}))

interface ICheckboxProps extends CheckboxProps {
  isError?: boolean
}

export const StyledCheckbox = styled((props: ICheckboxProps) => <Checkbox {...props} />, {
  shouldForwardProp: (prop) => prop !== 'isError',
})(({ theme, isError }) => ({
  color: isError ? theme.palette.error.main : '#dedede',
  '&:hover': {
    backgroundColor: 'transparent',
  },
  '&.Mui-disabled': {
    opacity: 0.2,
  },
  '.MuiSvgIcon-root': {
    transform: 'scale(1.25)',
  },
  '&.Mui-checked': {
    color: theme.palette.primary.main,
  },
}))
