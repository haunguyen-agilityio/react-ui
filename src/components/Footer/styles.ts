// Libraries
import { styled } from '@mui/system'
import ListItemButton from '@mui/material/ListItemButton'

// Themes
import theme from '../../themes'

export const StyledListItemButton = styled(ListItemButton)({
  textAlign: 'center',
  color: theme.palette.common.white,
  marginRight: theme.spacing(0.5),
  padding: theme.spacing(1.5, 0),
  '&:hover': {
    backgroundColor: theme.palette.grey[400],
    color: theme.palette.common.black,
  },
  '&.Mui-selected': {
    backgroundColor: theme.palette.grey[400],
    color: theme.palette.common.black,
  },
})

export const StyledFooter = styled('footer')({
  color: theme.palette.common.white,
  backgroundColor: theme.palette.grey[500],
  padding: theme.spacing(0, 4.2),
})
