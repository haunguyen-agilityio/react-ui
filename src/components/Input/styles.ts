// Libraries
import { styled } from '@mui/system'
import TextField from '@mui/material/TextField'
import FormHelperText from '@mui/material/FormHelperText'

// Themes
import theme from '@themes'
import typography from '@themes/typography'

export const InputWrapper = styled('div')({
  padding: theme.spacing(2),
})

export const StyledFormHelperText = styled(FormHelperText)({
  color: theme.palette.error.main,
  fontSize: typography.body1.fontSize,
})

export const StyledTextField = styled(TextField)((props) => ({
  '& .MuiInputBase-input': {
    color: props.error ? theme.palette.error.main : theme.palette.text.primary,
  },
  ' .MuiOutlinedInput-root': {
    '& fieldset': {
      borderRadius: 0,
      padding: theme.spacing(0),
      borderColor: props.error ? theme.palette.error.main : theme.palette.grey[400],
      '&.MuiOutlinedInput-notchedOutline': {
        borderColor: props.error ? theme.palette.error.main : theme.palette.grey[400],
      },
    },
    '&:hover fieldset': {
      border: props.error
        ? `1px solid ${theme.palette.error.main}`
        : `1px solid ${theme.palette.grey[400]}`,
    },
    '&.Mui-focused fieldset': {
      borderColor: theme.palette.grey[400],
      border: props.error
        ? `1px solid ${theme.palette.error.main}`
        : `1px solid ${theme.palette.grey[400]}`,
    },
  },
}))
