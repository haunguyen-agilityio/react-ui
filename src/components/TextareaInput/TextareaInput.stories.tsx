// Libraries
import React, { useState } from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import { Box } from '@mui/material'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '../../themes'

// Components
import TextareaInput from '.'

export default {
  title: 'Components/TextareaInput',
  component: TextareaInput,
  decorators: [
    (Story) => (
      <ThemeProvider theme={theme}>
        <Story />
      </ThemeProvider>
    ),
  ],
} as ComponentMeta<typeof TextareaInput>

export const Playground: ComponentStory<typeof TextareaInput> = (args) => {
  const [value, setValue] = useState('')

  const handleOnChange = (value: string) => {
    setValue(value)
  }

  return <TextareaInput {...args} value={value} onChange={handleOnChange} />
}

export const All = () => (
  <Box
    sx={{
      display: 'flex',
      flexDirection: 'column',
      rowGap: '20px',
    }}
  >
    <TextareaInput name='textarea-input-default' label='Default' value='' onChange={() => {}} />
    <TextareaInput
      name='textarea-input-error'
      label='Error'
      value=''
      onChange={() => {}}
      errorMessage='Required'
      error
    />
    <TextareaInput
      name='textarea-input-disabled'
      label='Disabled'
      value=''
      onChange={() => {}}
      disabled
    />
    <TextareaInput
      name='textarea-input-has-text'
      label='Has Text'
      value='Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
      onChange={() => {}}
    />
  </Box>
)

export const Default = () => (
  <TextareaInput name='textarea-input-default' label='Default' value='' onChange={() => {}} />
)

export const Error = () => (
  <TextareaInput
    name='textarea-input-error'
    label='Error'
    value=''
    onChange={() => {}}
    error
    errorMessage='Required'
  />
)

export const Disabled = () => (
  <TextareaInput
    name='textarea-input-disabled'
    label='Disabled'
    value=''
    onChange={() => {}}
    disabled
  />
)

export const HasText = () => (
  <TextareaInput
    name='textarea-input-has-text'
    label='Has Text'
    value='Lorem ipsum dolor sit amet, consectetur adipiscing elit. '
    onChange={() => {}}
  />
)
export {}
