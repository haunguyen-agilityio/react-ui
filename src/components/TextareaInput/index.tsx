// Libraries
import React, { memo } from 'react'
import { FormControl } from '@mui/material'

// Styles
import { StyledFormHelperText, StyledFormLabel, StyledTextField } from './styles'

export interface ITextareaInputProps {
  name: string
  value: string
  label?: string
  placeholder?: string
  disabled?: boolean
  error?: boolean
  errorMessage?: string
  fullWidth?: boolean
  rows?: number
  onChange: (value: string) => void
}

const TextareaInput = ({
  name = 'textarea-input-name',
  value = '',
  label = '',
  placeholder = 'Enter something',
  disabled = false,
  error = false,
  errorMessage = '',
  fullWidth = true,
  rows = 5,
  onChange = () => {},
}: ITextareaInputProps) => {
  const handleOnChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
    onChange(event.target.value)
  }

  return (
    <FormControl fullWidth={fullWidth} error={error}>
      {label && <StyledFormLabel disabled={disabled}>{label}</StyledFormLabel>}
      <StyledTextField
        name={name}
        value={value}
        placeholder={placeholder}
        disabled={disabled}
        onChange={handleOnChange}
        multiline
        rows={rows}
        error={error}
      />
      {error && <StyledFormHelperText>{errorMessage}</StyledFormHelperText>}
    </FormControl>
  )
}

export default memo(TextareaInput)
