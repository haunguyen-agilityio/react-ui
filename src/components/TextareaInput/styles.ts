// Libraries
import { FormHelperText, FormLabel, TextField } from '@mui/material'
import { styled } from '@mui/material/styles'

const StyledTextField = styled(TextField)(({ theme, error }) => ({
  '& .MuiOutlinedInput-input': {
    resize: 'vertical',
    fontSize: theme.typography.h4.fontSize,
  },
  '& .MuiOutlinedInput-root:hover fieldset': {
    borderColor: error ? theme.palette.error.main : theme.palette.grey[400],
  },
  '& .MuiOutlinedInput-root.Mui-focused fieldset': {
    borderColor: error ? theme.palette.error.main : theme.palette.grey[400],
  },
  '& .MuiInputBase-input::placeholder': {
    opacity: 1,
    color: error ? theme.palette.error.main : theme.palette.grey[400],
    fontSize: theme.typography.h4.fontSize,
  },
  '& fieldset': {
    borderRadius: 0,
  },
}))

const StyledFormHelperText = styled(FormHelperText)(({ theme }) => ({
  marginTop: theme.spacing(0.3),
  marginLeft: 0,
  fontSize: theme.typography.body1.fontSize,
}))

const StyledFormLabel = styled(FormLabel)(({ theme }) => ({
  fontSize: theme.typography.h5.fontSize,
  color: theme.palette.common.black,
}))

export { StyledFormHelperText, StyledFormLabel, StyledTextField }
