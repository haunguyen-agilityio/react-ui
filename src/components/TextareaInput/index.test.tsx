// Libraries
import { fireEvent, render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '../../themes'

// Components
import TextareaInput, { ITextareaInputProps } from '.'

describe('TextareaInput', () => {
  const componentProps: ITextareaInputProps = {
    name: 'test-textarea-name',
    onChange: jest.fn(),
    value: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  }
  const ThemedComponent = (props: ITextareaInputProps) => (
    <ThemeProvider theme={theme}>
      <TextareaInput {...props} />
    </ThemeProvider>
  )

  test('It should match DOM Snapshot when value is not empty', () => {
    const { asFragment } = render(<ThemedComponent {...componentProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('It should match DOM Snapshot when value is empty', () => {
    const { asFragment } = render(<ThemedComponent {...componentProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('It should match DOM Snapshot when error', () => {
    const { asFragment } = render(
      <ThemedComponent {...componentProps} error errorMessage='Required' />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('It should match DOM Snapshot when not full width', () => {
    const { asFragment } = render(<ThemedComponent {...componentProps} fullWidth={false} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('It should has placeholder', () => {
    const { getByPlaceholderText } = render(
      <ThemedComponent {...componentProps} value='' placeholder='Enter something' />,
    )
    expect(getByPlaceholderText('Enter something')).toBeTruthy()
  })

  test('It can be disabled', () => {
    const { getByDisplayValue } = render(<ThemedComponent {...componentProps} disabled />)
    expect(
      getByDisplayValue('Lorem ipsum dolor sit amet, consectetur adipiscing elit.'),
    ).toBeDisabled()
  })

  test('It should has label', () => {
    const { getByText } = render(<ThemedComponent {...componentProps} label='Textarea label' />)
    expect(getByText('Textarea label')).toBeTruthy()
  })

  test('It should trigger handleOnChange function when value changes', () => {
    const { getByDisplayValue } = render(<ThemedComponent {...componentProps} />)
    fireEvent.change(
      getByDisplayValue('Lorem ipsum dolor sit amet, consectetur adipiscing elit.'),
      { target: { value: 'Textarea value' } },
    )
    expect(componentProps.onChange).toHaveBeenCalledTimes(1)
  })
})
