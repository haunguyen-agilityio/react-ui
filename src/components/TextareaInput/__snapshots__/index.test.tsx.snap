// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextareaInput It should match DOM Snapshot when error 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-16bhqul-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-multiline css-rjyls5-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <textarea
          aria-invalid="true"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r2:"
          name="test-textarea-name"
          placeholder="Enter something"
          rows="5"
          style="height: 0px; overflow: hidden;"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </textarea>
        <textarea
          aria-hidden="true"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
          tabindex="-1"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-hdw1oc"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-19jrxw0-MuiFormHelperText-root"
    >
      Required
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextareaInput It should match DOM Snapshot when not full width 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-4hrhpy-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-rjyls5-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <textarea
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r3:"
          name="test-textarea-name"
          placeholder="Enter something"
          rows="5"
          style="height: 0px; overflow: hidden;"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </textarea>
        <textarea
          aria-hidden="true"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
          tabindex="-1"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-hdw1oc"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextareaInput It should match DOM Snapshot when value is empty 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-4hrhpy-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-rjyls5-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <textarea
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r1:"
          name="test-textarea-name"
          placeholder="Enter something"
          rows="5"
          style="height: 0px; overflow: hidden;"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </textarea>
        <textarea
          aria-hidden="true"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
          tabindex="-1"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-hdw1oc"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextareaInput It should match DOM Snapshot when value is not empty 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-4hrhpy-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-rjyls5-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <textarea
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          name="test-textarea-name"
          placeholder="Enter something"
          rows="5"
          style="height: 0px; overflow: hidden;"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </textarea>
        <textarea
          aria-hidden="true"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
          tabindex="-1"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-hdw1oc"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
